### DISCUSSION TITLE
HELP - I have many screenshots with text. How to implement OCR or integrate paperless-ngx for better image search?

### DISCUSSION THREAD

**Discussion by Fireflaker** [photoprism/photoprism#4011] [Ideas]
It looks like implementing OCR is no trivial task and is not on the roadmap😔. However, an OCR search capability similar to one in Google Photos is **_essential_** for me. I tried paperless-ngx, but it writes to the originals dataset and is impractical.

I wasn't able to find a solution online, and determined it is best to initiate a discussion thread here:
How can I implement a plugin or function to apply OCR to all photos when indexing, and append OCR text result to searchable index terms?

I was considering modifying the indexing process to send each photo to a locally running OCR API server (windows or Linux, on a different computer with better GPU), and adding the resulting text to properties; or using paperless-ngx to process OCR locally. May I request some guidance on where to get started?

Alternatively, are there better ways to approch this - like a readonly version of paperless-ngx that can run alongside photoprism in Docker/Truenas

**Comment by graciousgrey** [comment]
You can find a related issue here: https://github.com/photoprism/photoprism/issues/907 :)

**Comment by Fireflaker** [comment]
Thank you.
